lib/SILOptimizer/Mandatory/SendNonSendable.cpp

```diff
@@ -0,0 +1,33 @@
+#include "../../Sema/TypeCheckConcurrency.h"
+#include "swift/AST/Expr.h"
+#include "swift/SILOptimizer/PassManager/Transforms.h"
+
+using namespace swift;
+
+class SendNonSendable : public SILFunctionTransform {
+
+  // find any ApplyExprs in this function, and check if any of them make an
+  // unsatisfied isolation jump, emitting appropriate diagnostics if so
+  void run() override {
+    SILFunction *function = getFunction();
+
+    for (SILBasicBlock &bb : *function) {
+      for (SILInstruction &instr : bb) {
+        if (ApplyExpr *apply = instr.getLoc().getAsASTNode<ApplyExpr>()) {
+          if (apply->makesUnsatisfiedIsolationJump()) {
+            auto &ctx = instr.getModule().getASTContext();
+            if (ctx.LangOpts.hasFeature(Feature::DeferredSendableChecking)) {
+              diagnoseApplyArgSendability(apply);
+            }
```

Comment:
For efficiency, you need to check if this feature is enabled before looping over all of the instructions. You want to exit this method early if the feature isn't enabled. Otherwise this pass runs all the time, contributing to compilation time when not needed. The context of the instruction is the context of the function.

This fix in particular is definitely required before merging.